File: app/src/test/java/com/mgaetan89/showsrage/adapter/LogsAdapter_GetItemCountTest.java
package com.mgaetan89.showsrage.adapter;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;
import java.util.Collection;
import java.util.Collections;
import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(Parameterized.class)
public class LogsAdapter_GetItemCountTest {
	@Parameterized.Parameter(1)
	public int itemCount;

	@Parameterized.Parameter(0)
	public List<String> logs;

	private LogsAdapter adapter;

	@Before
	public void before() {
		this.adapter = new LogsAdapter(this.logs);
	}

	@Test
	public void getItemCount() {
		assertThat(this.adapter.getItemCount()).isEqualTo(this.itemCount);
	}

	@After
	public void after() {
		this.adapter = null;
	}

	@Parameterized.Parameters
	public static Collection<Object[]> data() {
		return Arrays.asList(new Object[][]{
				{Collections.emptyList(), 0},
				{Collections.singletonList(""), 1},
				{Arrays.asList("", "", ""), 3},
		});
	}
}

File: app/src/test/kotlin/com/mgaetan89/showsrage/adapter/LogsAdapter_GetItemCountTest.kt
package com.mgaetan89.showsrage.adapter

import com.mgaetan89.showsrage.model.LogEntry
import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

@RunWith(Parameterized::class)
class LogsAdapter_GetItemCountTest(val logs: List<LogEntry>, val itemCount: Int) {
    private lateinit var adapter: LogsAdapter

    @Before
    fun before() {
        this.adapter = LogsAdapter(this.logs)
    }

    @Test
    fun getItemCount() {
        assertThat(this.adapter.itemCount).isEqualTo(this.itemCount)
    }

    companion object {
        @JvmStatic
        @Parameterized.Parameters
        fun data(): Collection<Array<Any>> {
            return listOf(
                    arrayOf(emptyList<Any>(), 0),
                    arrayOf(listOf(LogEntry()), 1),
                    arrayOf(listOf(LogEntry(), LogEntry(), LogEntry()), 3)
            )
        }
    }
}
